experiments/gromacs/rocm/ramble.yaml

```diff
@@ -53,10 +53,6 @@ ramble:
     environments:
       gromacs:
         packages:
-        - hip543
```

Comment:
This works for rocm/5.4.3. However, if I change the compiler to rocmcc@5.5.1, the spack concretizer still uses v5.4.3 for some packages. Following is the concretizer output for rocmcc@5.5.1

```
==> Concretized gromacs@2023.3%rocmcc@5.5.1~double~hwloc+mpi+openmp+rocm+sycl amdgpu_target=gfx90a
 -   2fjfk3y  gromacs@2023.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cp2k~cuda~cycle_subcounters~double+gmxapi~heffte~hwloc~intel-data-center-gpu-max~intel_provided_gcc~ipo~mdrun_only+mpi+nblib~nosuffix~opencl+openmp~plumed~relaxed_double_precision+rocm+shared+sycl amdgpu_target=gfx90a build_system=cmake build_type=Release generator=make openmp_max_threads=none arch=linux-rhel8-x86_64
[e]  jmwvqsy      ^cmake@3.23.1%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rhel8-x86_64
[e]  elntlub      ^comgr@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2'  build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[e]  v42qoop      ^cray-mpich@8.1.26%cce@17.0.0~gtl+wrappers build_system=generic arch=linux-rhel8-x86_64
[e]  6nfuiwx      ^gmake@4.2.1%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~guile build_system=generic patches=ca60bd9,fe5b60d arch=linux-rhel8-x86_64
[e]  43ygv4w      ^hip@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cuda+rocm build_system=cmake build_type=Release generator=make patches=5068750,c2ee21c,ca523f1,ddd86f0 arch=linux-rhel8-x86_64
 -   uisys2n      ^hipsycl@0.9.4%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cuda~ipo+rocm amdgpu_target=gfx90a build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
 -   rzxtvzu          ^boost@1.69.0%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~atomic~chrono~clanglibcpp~container+context~contract~coroutine~date_time~debug~exception+fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace~system~taggedlayout~test~thread~timer~type_erasure~versionedlayout~wave build_system=generic context-impl=fcontext cxxstd=17 patches=3a83d90,5da7ad2,607b077 visibility=hidden arch=linux-rhel8-x86_64
[e]  goajfe5          ^python@3.9.12%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' +bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,4c24573,ebdca64,f2fd060 arch=linux-rhel8-x86_64
[e]  qd5rbts      ^hsa-rocr-dev@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~asan+image+shared build_system=cmake build_type=Release generator=make patches=9267179 arch=linux-rhel8-x86_64
[e]  3f7dz3g      ^hsakmt-roct@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~asan+shared build_system=cmake build_type=Release generator=make patches=114b05a arch=linux-rhel8-x86_64
[e]  e4ljwkx      ^intel-oneapi-mkl@2023.2.0%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cluster+envmods~ilp64+shared build_system=generic mpi_family=none threads=none arch=linux-rhel8-x86_64
[e]  5554olq      ^llvm-amdgpu@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=a08bbe1 arch=linux-rhel8-x86_64
[e]  tsvmocc      ^rocminfo@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2'  build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[e]  6bvget5      ^roctracer-dev-api@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2'  build_system=generic arch=linux-rhel8-x86_64

==> Concretized hipsycl@0.9.4%rocmcc@5.5.1+rocm
 -   uisys2n  hipsycl@0.9.4%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cuda~ipo+rocm amdgpu_target=gfx90a build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
 -   rzxtvzu      ^boost@1.69.0%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~atomic~chrono~clanglibcpp~container+context~contract~coroutine~date_time~debug~exception+fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace~system~taggedlayout~test~thread~timer~type_erasure~versionedlayout~wave build_system=generic context-impl=fcontext cxxstd=17 patches=3a83d90,5da7ad2,607b077 visibility=hidden arch=linux-rhel8-x86_64
[e]  jmwvqsy      ^cmake@3.23.1%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rhel8-x86_64
[e]  6nfuiwx      ^gmake@4.2.1%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~guile build_system=generic patches=ca60bd9,fe5b60d arch=linux-rhel8-x86_64
[e]  43ygv4w      ^hip@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cuda+rocm build_system=cmake build_type=Release generator=make patches=5068750,c2ee21c,ca523f1,ddd86f0 arch=linux-rhel8-x86_64
[e]  qd5rbts      ^hsa-rocr-dev@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~asan+image+shared build_system=cmake build_type=Release generator=make patches=9267179 arch=linux-rhel8-x86_64
[e]  5554olq      ^llvm-amdgpu@5.4.3%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=a08bbe1 arch=linux-rhel8-x86_64
[e]  goajfe5      ^python@3.9.12%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' +bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,4c24573,ebdca64,f2fd060 arch=linux-rhel8-x86_64

==> Concretized intel-oneapi-mkl@2023.2.0
[e]  e4ljwkx  intel-oneapi-mkl@2023.2.0%rocmcc@5.5.1 cflags='-g -O2' cxxflags='-g -O2' ~cluster+envmods~ilp64+shared build_system=generic mpi_family=none threads=none arch=linux-rhel8-x86_64

```

Reply:
Thanks: the components were synchronized with each other WRT version, but not with the compiler version. I added another constraint for that in https://github.com/LLNL/benchpark/pull/150/commits/1f16b8b3e5deabc31ac869323f12d97f068c81cb  So when building with `%rocmcc@5.4.3`, we will use `^hip@5.4.3`.

I'll note that with this change specifically, Benchpark constraints for rocmcc/rocm-externals are more strict than within the non-external Spack builtin case.

Reply:
Would this still work for a non-rocmcc compiler, say cce?

Reply:
Yes: this does not force the use of `%rocmcc`, it just makes sure that if you use `%rocmcc`, that the compiler version matches the component version.

Reply:
The spack concretization works for consistently for both rocm 5.4 and 5.5